src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1525,68 +1556,449 @@ static void LoadRegPair(BYTE* p, int reg1, int reg2, UINT32 offset)
+            {
+                codeSize += (pLookup->sizeOffset > 2047 ? 28 : 16);
+                indirectionsDataSize += (pLookup->sizeOffset > 2047 ? 4 : 0);
+            }
+
+            codeSize += (pLookup->offsets[i] > 2047 ? 24 : 16); // if( > 2047) (24 bytes) else 16 bytes for instructions.
+            indirectionsDataSize += (pLookup->offsets[i] > 2047 ? 4 : 0); // 4 bytes for storing indirection offset values
+        }
+
+        codeSize += indirectionsDataSize ? 4 : 0; // auipc
+
+        if (pLookup->testForNull)
+        {
+            codeSize += 12; // beq-ret-addi
+
+            //padding for 8-byte align (required by EmitHelperWithArg)
+            codeSize += ALIGN_UP(codeSize, 8);
+
+            codeSize += 32; // size of EmitHelperWithArg
+        }
+        else
+        {
+            codeSize += 4; /* jalr */
+        }
+
+        // the offset value of data_label.
+        uint dataOffset = codeSize;
+
+        codeSize += indirectionsDataSize;
+
+        BEGIN_DYNAMIC_HELPER_EMIT(codeSize);
+
+        BYTE * old_p = p;
+
+        if (indirectionsDataSize)
+        {
+            _ASSERTE(codeSize < 2047);
+
+            //auipc t4, 0
+            *(DWORD*)p = UTypeInstr(0x17, RegT4, 0);
+            p += 4;
+        }
+
+        if (pLookup->testForNull || pLookup->sizeOffset != CORINFO_NO_SIZE_CHECK)
+        {
+            *(DWORD*)p = ITypeInstr(0x13, 0, RegT2, RegA0, 0);// addi t2, a0, 0
+            p += 4;
+        }
+
+        BYTE* pBLTCall = NULL;
+
+        for (WORD i = 0; i < pLookup->indirections; i++)
+        {
+            if (i == pLookup->indirections - 1 && pLookup->sizeOffset != CORINFO_NO_SIZE_CHECK)
+            {
+                _ASSERTE(pLookup->testForNull && i > 0);
+
+                if (pLookup->sizeOffset > 2047)
+                {
+                    *(DWORD*)p = ITypeInstr(0x3, 0x2, RegT4, RegT4, dataOffset);// lw  t4, dataOffset(t4)
+                    p += 4;
+                    *(DWORD*)p = RTypeInstr(0x33, 0, 0, RegT5, RegA0, RegT4);// add  t5, a0, t4
+                    p += 4;
+                    *(DWORD*)p = ITypeInstr(0x3, 0x3, RegT5, RegT5, 0);// ld  t5, 0(t5)
+                    p += 4;
+
+                    // move to next indirection offset data
+                    dataOffset += 4;
+                }
+                else
+                {
+                    *(DWORD*)p = ITypeInstr(0x3, 0x3, RegT5, RegA0, (UINT32)pLookup->sizeOffset);// ld  t5, #(pLookup->sizeOffset)(a0)
+                    p += 4;
+                }
+                // lui  t4, (slotOffset&0xfffff000)>>12
+                *(DWORD*)p = UTypeInstr(0x37, RegT4, (((UINT32)slotOffset & 0xfffff000) >> 12));
+                p += 4;
+                *(DWORD*)p = ITypeInstr(0x13, 0, RegT4, RegT4, slotOffset & 0xfff);// addi  t4, t4, (slotOffset&0xfff)
+                p += 4;
+                // blt  t4, t5, CALL HELPER
+                pBLTCall = p;       // Offset filled later
```

Comment:
p += 4; should not be removed. Please add p += 4;

Reply:
Done